### PR TITLE
Remove unnecessary queries to integration entity table

### PR DESCRIPTION
**Pull request by dadarya0**
| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | No
| New feature/enhancement? (use the a.x branch)      | Yes
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | No
| Automated tests included?              | Yes
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 

#### Description:
Stop executing unnecessary queries to the integration_entity table on add/update lead if lead sync related plugins are not enabled.

https://github.com/mautic/mautic/blob/5.x/app/bundles/PluginBundle/EventListener/LeadSubscriber.php#L61
https://github.com/mautic/mautic/blob/5.x/app/bundles/PluginBundle/Entity/IntegrationEntityRepository.php#L516

#### Steps to test this PR:

1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Disable all lead sync related plugin
3. Unnecessary delete/update queries to the integration_entity table on add/update lead should not be executed.

### CODE REVIEW

**Comment by kuzmany** [comment]
@dadarya0 this is ready to tests or WIP?

**Comment by dadarya0** [comment]
> @dadarya0 this is ready to tests or WIP?

it is ready to test.

File: app/bundles/PluginBundle/EventListener/LeadSubscriber.php [hunk 1]
```diff
@@ -29,7 +35,7 @@ public static function getSubscribedEvents(): array
      */
     public function onLeadDelete(LeadEvent $event): bool
     {
-        /** @var \Mautic\LeadBundle\Entity\Lead $lead */
+        /** @var Lead $lead */
         $lead                  = $event->getLead();
         $integrationEntityRepo = $this->pluginModel->getIntegrationEntityRepository();
         $integrationEntityRepo->findLeadsToDelete('lead%', $lead->getId());
```
Comment (kuzmany) [inline app/bundles/PluginBundle/EventListener/LeadSubscriber.php:41]: Shouldn't we do same here? Same  in onCompanyDelete

**Review by kuzmany — CHANGES_REQUESTED** [review]

File: app/bundles/PluginBundle/EventListener/LeadSubscriber.php (hunk 1 above)
Comment (kuzmany) [reply app/bundles/PluginBundle/EventListener/LeadSubscriber.php:41]: @dadarya0 did you have time to check it?

Comment (dadarya0) [reply app/bundles/PluginBundle/EventListener/LeadSubscriber.php:41]: @kuzmany we should not restrict record deletion as I can explain in below example.

- Configure any relevant plugin, enable sync
- After sync some record will be added in `integration_entity` table
- Now disable/unpublish plugin
- No if we are restricting deletion of record in `integration_entity` table then those record will be available forever and those are useless and unmapped as lead/company is already deleted.

I believe that now it is clear the reason of not implementing same logic in `onCompanyDelete` and `onLeadDelete`

Comment (kuzmany) [reply app/bundles/PluginBundle/EventListener/LeadSubscriber.php:41]: Thanks. Sounds clear 🥇

**Review by kuzmany — APPROVED** [review]
Works for me 👍

**Review by escopecz — APPROVED** [review]
I see no issue in the code. Thank you Saurabh!

**Pull request merged**